### PR TITLE
Target VM name length-related changes

### PR DESCRIPTION
**Pull request by jakub-dzon**
Re: https://bugzilla.redhat.com/show_bug.cgi?id=1857165

Series of related fixes:
- Failing import that encounters problems with DV creation (including adding volume and disk to a VM);
- Rejecting (on API level) CRs with targetVMName longer than 63 characters;
- Shortening target VM name to 63 characters.;
- Shortening VM's disk and volume names to 63 characters.;

```release-note
NONE
```

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -123,17 +120,16 @@ func WithMessage(message string, newMessage string) string {
 	return fmt.Sprintf("%s, %s", message, newMessage)
 }
 
-// MakeLabelFrom creates label value from given CR name
-func MakeLabelFrom(name string) string {
-	n := len(name)
-	if n > MaxLabelValueLength {
-		log.Info("Label value will be shortened to 63 characters", "CR name", name)
+// EnsureLabelValueLength shortens given value to the maximal label value length of 63 characters
```
Comment (pkliczewski) [inline pkg/utils/utils.go]: s/maximal/maximum

File: pkg/providers/ovirt/mapper/mapper.go [hunk 2]
```diff
@@ -391,6 +393,17 @@ func (o *OvirtMapper) mapNetworkType(mapping v2vv1alpha1.ResourceMappingItem, ku
 
 // ResolveVMName resolves the target VM name
 func (o *OvirtMapper) ResolveVMName(targetVMName *string) *string {
+	vmNameBase := o.resolveVMNameBase(targetVMName)
+	if vmNameBase == nil {
+		return nil
+	}
+	// VM name is put in label values and has to be shorter than regular k8s name
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1857165
```
Comment (pkliczewski) [inline pkg/providers/ovirt/mapper/mapper.go]: No need to add `show_bug.cgi?id=` in the url

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go [hunk 3]
```diff
@@ -725,23 +732,8 @@ func (r *ReconcileVirtualMachineImport) createDataVolume(provider provider.Provi
 	err = r.client.Create(context.TODO(), &dv)
 	if err != nil {
 		message := fmt.Sprintf("Data volume %s/%s creation failed: %s", dv.Namespace, dv.Name, err)
-		// Update event:
-		r.recorder.Event(instance, corev1.EventTypeWarning, EventDVCreationFailed, message)
-
-		// Update condition to failed:
-		succeededCond := conditions.NewSucceededCondition(string(v2vv1alpha1.DataVolumeCreationFailed), message, corev1.ConditionFalse)
```
Comment (pkliczewski) [inline pkg/controller/virtualmachineimport/virtualmachineimport_controller.go:732]: any reason why we do not need this condition any more?

Comment (jakub-dzon) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go:732]: It is added on the caller level @ line 356.

Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go:732]: OK

File: pkg/utils/utils.go (hunk 1 above)
Comment (jakub-dzon) [reply pkg/utils/utils.go]: Done

File: pkg/providers/ovirt/mapper/mapper.go (hunk 2 above)
Comment (jakub-dzon) [reply pkg/providers/ovirt/mapper/mapper.go]: Done

**Review by pkliczewski — APPROVED** [review]

**Review by machacekondra — APPROVED** [review]

**Comment by masayag** [comment]
This PR seems to be a good candidate for a backport to v0.1 branch.

**Pull request merged**